newsroom/search/service.py

```diff
@@ -511,21 +511,33 @@ def prefill_search_items(self, search):
             )
 
     def prefill_search_highlights(self, search, req):
-        query_string = search.args.get("q")
-        query_string_settings = app.config["ELASTICSEARCH_SETTINGS"]["settings"]["query_string"]
+        query = search.args.get("q")
         advanced_search = json.loads(search.args.get("advanced", "{}"))
 
         if app.data.elastic.should_highlight(req) and (
-            query_string or advanced_search.get("all") or advanced_search.get("any")
+            query or advanced_search.get("all") or advanced_search.get("any")
         ):
-            selected_field = advanced_search.get("fields", [])
+            selected_fields = advanced_search.get("fields", [])
+
+            should_queries = []
+            if query:
+                should_queries.append(query_string(query))
+            all_query = advanced_search.get("all")
+            if all_query:
+                should_queries.append(query_string(all_query))
+            any_query = advanced_search.get("any")
+            if any_query:
+                should_queries.append(query_string(any_query))
+
+            search.query.setdefault("bool", {})
+            search.query["bool"]["should"] = should_queries
```

Comment:
there is a helper to create such query, would be better to use that one:
https://github.com/superdesk/newsroom-core/blob/develop/newsroom/search/service.py#L708

Reply:
Ok I'll check